sql: panic in selectNamesAndTemplates due to makeslice: cap out of range

roachtest.sqlsmith/setup=seed/setting=no-ddl [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8240112?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8240112?buildTab=artifacts#/sqlsmith/setup=seed/setting=no-ddl) on master @ [27ebfa830ac7122fc3b286136158fef615b20b8f](https://github.com/cockroachdb/cockroach/commits/27ebfa830ac7122fc3b286136158fef615b20b8f):


```
test artifacts and logs in: /artifacts/sqlsmith/setup=seed/setting=no-ddl/run_1
(sqlsmith.go:276).func3: ping node 1: dial tcp 35.185.34.122:26257: connect: connection refused
HINT: node likely crashed, check logs in artifacts > logs/1.unredacted
previous sql:
SELECT
	crdb_internal.generate_test_objects('X':::STRING::STRING, 8248030903138497987:::INT8::INT8)::JSONB AS col_8841,
	'{"8lEyMTh9": null, "baz": null, "c": {"bar": "m9LH", "c": null}}':::JSONB AS col_8842
FROM
	defaultdb.public.seed AS tab_3698 JOIN defaultdb.public.seed AS tab_3699 ON (tab_3698._jsonb) = (tab_3699._jsonb)
WHERE
	true;ping node 1: dial tcp 35.185.34.122:26257: connect: connection refused
HINT: node likely crashed, check logs in artifacts > logs/1.unredacted
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed/setting=no-ddl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23219

There was a panic on node 1:
```
panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range [recovered]
	panic: runtime error: makeslice: cap out of range

goroutine 3456 [running]:
panic({0x509c300, 0xc007d3dda0})
	GOROOT/src/runtime/panic.go:987 +0x3ba fp=0xc005163a88 sp=0xc0051639c8 pc=0x49dd5a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc002c29000, {0x6b13100, 0xc002038e40}, {0x4ddc8e0?, 0x6aca960?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1080 +0x1b4 fp=0xc005163b18 sp=0xc005163a88 pc=0x31f6954
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0x6b13100, 0xc002038e40}, {0xf011cf636?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:832 +0x4a fp=0xc005163b58 sp=0xc005163b18 pc=0x31f494a
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func3()
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:833 +0x32 fp=0xc005163b80 sp=0xc005163b58 pc=0x31f4b52
runtime.deferCallSave(0xc005163c50, 0xc005169df0?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc005163b90 sp=0xc005163b80 pc=0x49d948
runtime.runOpenDeferFrame(0xc00266b900?, 0xc0027fee60)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc005163bd8 sp=0xc005163b90 pc=0x49d765
panic({0x4ddc8e0, 0x6aca960})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc005163c98 sp=0xc005163bd8 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc005163e38 sp=0xc005163c98 pc=0x1bb91e7
runtime.deferCallSave(0xc005163f08, 0xc005167788?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc005163e48 sp=0xc005163e38 pc=0x49d948
runtime.runOpenDeferFrame(0xc00266b900?, 0xc0027fee60)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc005163e90 sp=0xc005163e48 pc=0x49d765
panic({0x4ddc8e0, 0x6aca960})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc005163f50 sp=0xc005163e90 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc0051640f0 sp=0xc005163f50 pc=0x1bb91e7
runtime.deferCallSave(0xc0051641c0, 0xc0051676d0?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc005164100 sp=0xc0051640f0 pc=0x49d948
runtime.runOpenDeferFrame(0xc00266a780?, 0xc00266b900)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc005164148 sp=0xc005164100 pc=0x49d765
panic({0x4ddc8e0, 0x6aca960})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc005164208 sp=0xc005164148 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc0051643a8 sp=0xc005164208 pc=0x1bb91e7
runtime.deferCallSave(0xc005164478, 0xc005165c30?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc0051643b8 sp=0xc0051643a8 pc=0x49d948
runtime.runOpenDeferFrame(0xc00266b900?, 0xc00266a780)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc005164400 sp=0xc0051643b8 pc=0x49d765
panic({0x4ddc8e0, 0x6aca960})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc0051644c0 sp=0xc005164400 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc005164660 sp=0xc0051644c0 pc=0x1bb91e7
runtime.deferCallSave(0xc005164730, 0xc005165b60?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc005164670 sp=0xc005164660 pc=0x49d948
runtime.runOpenDeferFrame(0xc0027fee60?, 0xc00266b900)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc0051646b8 sp=0xc005164670 pc=0x49d765
panic({0x4ddc8e0, 0x6aca960})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc005164778 sp=0xc0051646b8 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).Generate.func1.1()
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgen.go:148 +0x78 fp=0xc0051647a0 sp=0xc005164778 pc=0x300e238
runtime.deferCallSave(0xc005164870, 0xc005164ec0?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc0051647b0 sp=0xc0051647a0 pc=0x49d948
runtime.runOpenDeferFrame(0xc00266b900?, 0xc0027fee60)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc0051647f8 sp=0xc0051647b0 pc=0x49d765
panic({0x4ddc8e0, 0x6aca960})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc0051648b8 sp=0xc0051647f8 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).genMultipleTables.func1()
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/tables.go:42 +0x2ef fp=0xc0051649c0 sp=0xc0051648b8 pc=0x300fe2f
runtime.deferCallSave(0xc005164a90, 0xc005164d48?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc0051649d0 sp=0xc0051649c0 pc=0x49d948
runtime.runOpenDeferFrame(0xc005164c48?, 0xc00266b900)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc005164a18 sp=0xc0051649d0 pc=0x49d765
panic({0x4ddc8e0, 0x6aca960})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc005164ad8 sp=0xc005164a18 pc=0x49dbb2
runtime.panicmakeslicecap(...)
	GOROOT/src/runtime/slice.go:33
runtime.makeslice(0x0?, 0x0?, 0xc005164c58?)
	GOROOT/src/runtime/slice.go:100 +0xa5 fp=0xc005164b00 sp=0xc005164ad8 pc=0x4b6005
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).selectNamesAndTemplates(0xc00bd20700, {0x6b131a8, 0xc008653830}, 0x6ba6db0?, 0x7276e4680e03b5c3)
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/tables.go:98 +0xd8 fp=0xc005164c78 sp=0xc005164b00 pc=0x300ff38
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).genMultipleTables(0xc00bd20700, {0x6b131a8, 0xc008653830}, {0x6ba6db0?, 0xc004f0e0b0?}, {0x6ba51f0, 0xc0069e54a0})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/tables.go:58 +0x225 fp=0xc005164d90 sp=0xc005164c78 pc=0x300f8a5
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).maybeGenSchemas(0xc00bd20700?, {0x6b131a8?, 0xc008653830?}, 0x1?)
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/recurse.go:49 +0xa9 fp=0xc005164df0 sp=0xc005164d90 pc=0x300ed89
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).maybeGenDatabases(0xc00bd20700, {0x6b131a8, 0xc008653830})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/recurse.go:32 +0x6f fp=0xc005164e40 sp=0xc005164df0 pc=0x300ec2f
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).Generate.func1(0xc00bd20700, 0x1, {0x6b131a8, 0xc008653830})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgen.go:189 +0x32f fp=0xc005164f58 sp=0xc005164e40 pc=0x300df2f
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).Generate(0xc00bd20700, {0x6b131a8, 0xc008653830})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgen.go:204 +0x133 fp=0xc005164fa8 sp=0xc005164f58 pc=0x300da33
github.com/cockroachdb/cockroach/pkg/sql.(*planner).GenerateTestObjects(0xc007cf0000, {0x6b131a8, 0xc008653830}, {0xc004ddccc0, 0x2f})
	github.com/cockroachdb/cockroach/pkg/sql/generate_objects.go:65 +0x5b5 fp=0xc005165378 sp=0xc005164fa8 pc=0x330a595
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.glob..func291({0x6b131a8, 0xc008653830}, 0xc00d48f500, {0xc006a557d0?, 0xc65440?, 0xc005165698?})
	github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins.go:6383 +0x90 fp=0xc0051653b0 sp=0xc005165378 pc=0x1c65330
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.(*evaluator).EvalFuncExpr(0xc00d48f500, {0x6b131a8, 0xc008653830}, 0xc00592c400)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:479 +0x118 fp=0xc005165648 sp=0xc0051653b0 pc=0x18b5838
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FuncExpr).Eval(0xc008653440?, {0x6b131a8?, 0xc008653830?}, {0x6ba4840?, 0xc00d48f500?})
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/sem/tree/eval_expr_generated.go:276 +0x37 fp=0xc005165678 sp=0xc005165648 pc=0x13cbe97
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.Expr(...)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:26
github.com/cockroachdb/cockroach/pkg/sql.(*valuesNode).startExec(0xc007a521e0, {{0x6b131a8, 0xc008653830}, 0xc00d48f500, 0xc007cf0000})
	github.com/cockroachdb/cockroach/pkg/sql/values.go:80 +0x342 fp=0xc0051657e8 sp=0xc005165678 pc=0x34490e2
github.com/cockroachdb/cockroach/pkg/sql.startExec.func2({0xc0051658d8?, 0x473284?}, {0x6b160b0, 0xc007a521e0})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:523 +0x143 fp=0xc005165898 sp=0xc0051657e8 pc=0x3381b03
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112 +0x3e fp=0xc0051658d0 sp=0xc005165898 pc=0x346243e
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc0051659e8, {0x6b160b0?, 0xc007a521e0?}, {0x55a9a62?, 0x6?})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297 +0x14c3 fp=0xc005165960 sp=0xc0051658d0 pc=0x3462343
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc0051659e8, {0x6b160b0, 0xc007a521e0})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79 +0xf7 fp=0xc0051659b0 sp=0xc005165960 pc=0x3460d77
github.com/cockroachdb/cockroach/pkg/sql.walkPlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
github.com/cockroachdb/cockroach/pkg/sql.startExec({{0x6b131a8?, 0xc008653830?}, 0xc00d48f500?, 0xc007cf0000?}, {0x6b160b0, 0xc007a521e0})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:526 +0x119 fp=0xc005165a30 sp=0xc0051659b0 pc=0x3381959
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start(0xc003078300, {0x6b131a8, 0xc008653800})
	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:173 +0xbd fp=0xc005165a70 sp=0xc005165a30 pc=0x33850dd
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Init(0xc006382c00, {0x6b131a8, 0xc0086537d0})
	github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:187 +0xea fp=0xc005165aa8 sp=0xc005165a70 pc=0x26d09ea
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*panicInjector).Init(0xc0086536e0, {0x6b131a8?, 0xc0086537d0?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/panic_injector.go:62 +0xb7 fp=0xc005165b20 sp=0xc005165aa8 pc=0x2f5eed7
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:260 +0x34 fp=0xc005165b48 sp=0xc005165b20 pc=0x26d4df4
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc007d0cf00?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc005165b88 sp=0xc005165b48 pc=0x1bb8d82
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Start(0xc007d0cf00, {0x6b131a8?, 0xc0086537a0?})
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:259 +0x7a fp=0xc005165bf0 sp=0xc005165b88 pc=0x26d4d3a
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:122 +0x34 fp=0xc005165c18 sp=0xc005165bf0 pc=0x2f5dd34
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc005209180?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc005165c58 sp=0xc005165c18 pc=0x1bb8d82
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start(0xc005209180, {0x6b13100?, 0xc006eddf40?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:121 +0x70 fp=0xc005165cb0 sp=0xc005165c58 pc=0x2f5dcb0
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc005209180, {0x6b13100?, 0xc006eddf40?})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:732 +0x3a fp=0xc005165cf0 sp=0xc005165cb0 pc=0x1ee289a
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc007cba600, {0x6b13100, 0xc006eddf40})
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:483 +0x254 fp=0xc005165db0 sp=0xc005165cf0 pc=0x24127d4
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc00d48ee00?, {0x6b13100?, 0xc006eddf40?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:296 +0x205 fp=0xc005165e50 sp=0xc005165db0 pc=0x2f639c5
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc000aaaf00, {0x6b131a8, 0xc0086532c0}, 0xc00592c480, 0xc008a4a580, 0xc00592c500, 0xc000a71500, 0xc00d48ee00, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:827 +0x97b fp=0xc0051665a0 sp=0xc005165e50 pc=0x32c281b
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0x6b131a8?, {0x6b131a8, 0xc0086532c0}, 0xc007cf0000?, 0xc00592c480, 0xc007a521e0?, {{0x6b160b0?, 0xc007a521e0?}, 0x0?}, 0xc000a71500)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1795 +0x1f6 fp=0xc005166658 sp=0xc0051665a0 pc=0x32c70f6
github.com/cockroachdb/cockroach/pkg/sql.runPlanInsidePlan({0x6b131a8, 0xc0086532c0}, {{0x6b131a8, 0xc0086532c0}, 0xc002c298b0, 0xc002c29540}, 0xc0042367e0, {0x6b14088?, 0xc0070bf7c0?})
	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:334 +0x873 fp=0xc005166e10 sp=0xc005166658 pc=0x31c3f93
github.com/cockroachdb/cockroach/pkg/sql.(*planner).EvalRoutineExpr.func2(0xc003462e10, 0xc005166f30, {0x6b131a8, 0xc0086525d0}, 0xc002c298b0?, {0xc0070bf7a0, 0x2, 0x2}, 0xc0070bf7c0, 0xc008a4a580, ...)
	github.com/cockroachdb/cockroach/pkg/sql/routine.go:100 +0x276 fp=0xc005166ed0 sp=0xc005166e10 pc=0x33c13f6
github.com/cockroachdb/cockroach/pkg/sql.(*planner).EvalRoutineExpr(0xc002c29540, {0x6b131a8?, 0xc0086525d0}, 0xc003462e10, {0xc0070bf7a0?, 0x2, 0x2})
	github.com/cockroachdb/cockroach/pkg/sql/routine.go:106 +0x41c fp=0xc005167038 sp=0xc005166ed0 pc=0x33c0c3c
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.(*evaluator).EvalRoutineExpr(0xc005b13800?, {0x6b131a8?, 0xc0086525d0?}, 0xc003462e10?)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:635 +0xaa fp=0xc0051670a8 sp=0xc005167038 pc=0x18b684a
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*RoutineExpr).Eval(0x4e6aa00?, {0x6b131a8?, 0xc0086525d0?}, {0x6ba4840?, 0xc005b13800?})
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/sem/tree/eval_expr_generated.go:346 +0x37 fp=0xc0051670d8 sp=0xc0051670a8 pc=0x13cc597
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.Expr(...)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:26
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb.(*ExprHelper).Eval(0xc006702000, {0x6b131a8?, 0xc0086525d0?}, {0xc007aae000?, 0x0?, 0xc005167180?})
	github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/pkg/sql/execinfrapb/expr.go:248 +0x192 fp=0xc005167140 sp=0xc0051670d8 pc=0x1e8ca52
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcOutputHelper).ProcessRow(0xc00680e368, {0x6b131a8, 0xc0086525d0}, {0xc007aae000, 0x2, 0x2})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:275 +0x18e fp=0xc005167200 sp=0xc005167140 pc=0x1ee128e
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBase).ProcessRowHelper(0xc00680e280, {0xc007aae000?, 0xc0070bf210?, 0x1200000040?})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:711 +0x5c fp=0xc005167258 sp=0xc005167200 pc=0x1ee279c
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*noopProcessor).Next(0xc00680e280)
	github.com/cockroachdb/cockroach/pkg/sql/rowexec/noop.go:103 +0x65 fp=0xc005167290 sp=0xc005167258 pc=0x2186cc5
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Next(0xc002fc4600)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:243 +0x13c fp=0xc005167610 sp=0xc005167290 pc=0x26d0cdc
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*panicInjector).Next(0xc008652480)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/panic_injector.go:70 +0x7e fp=0xc005167688 sp=0xc005167610 pc=0x2f5f03e
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(0xc007bb5e00)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:116 +0x29 fp=0xc0051676a0 sp=0xc005167688 pc=0x2f61f09
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next-fm()
	<autogenerated>:1 +0x26 fp=0xc0051676b8 sp=0xc0051676a0 pc=0x2f6d186
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc006edd940?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc0051676f8 sp=0xc0051676b8 pc=0x1bb8d82
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0xc007bb5e00)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:124 +0x52 fp=0xc005167740 sp=0xc0051676f8 pc=0x2f61fb2
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).nextAdapter(0xc00b813220)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:252 +0x2a fp=0xc005167758 sp=0xc005167740 pc=0x2f5e60a
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).nextAdapter-fm()
	<autogenerated>:1 +0x26 fp=0xc005167770 sp=0xc005167758 pc=0x2f6cb26
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc005568400?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc0051677b0 sp=0xc005167770 pc=0x1bb8d82
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).next(...)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:256
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).Run(0xc00b813220, {0x6b13100?, 0xc006edd600?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:294 +0x2e5 fp=0xc005167928 sp=0xc0051677b0 pc=0x2f5ea25
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc008693380, {0x6b13100, 0xc006edd600})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:309 +0x1d8 fp=0xc0051679c8 sp=0xc005167928 pc=0x2f63998
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc000aaaf00, {0x6b131a8, 0xc008615800}, 0xc0058a7b80, 0xc008a4a580, 0xc0058a7e80, 0xc0086c3500, 0xc002c298b0, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:827 +0x97b fp=0xc005168118 sp=0xc0051679c8 pc=0x32c281b
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc008615860?, {0x6b131a8, 0xc008615800}, 0x0?, 0xc0058a7b80, 0xc008615800?, {{0x6b15860?, 0xc0086c9680?}, 0x0?}, 0xc0086c3500)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1795 +0x1f6 fp=0xc0051681d0 sp=0xc005168118 pc=0x32c70f6
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc0058a7b80?, {0x6b131a8, 0xc008615800}, 0xc002c29540?, 0x1?, 0xc002c29540, 0xc0086c3500, 0xc00026a1e0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1547 +0x206 fp=0xc0051682e8 sp=0xc0051681d0 pc=0x32c5506
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc002c29000, {0x6b131a8, 0xc008615800}, 0xc002c29540, 0xc008615800?, {0x7f525b325e98?, 0xc00096ef20}, 0x0?, 0xc007797598)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1633 +0x4ac fp=0xc0051683f0 sp=0xc0051682e8 pc=0x32113cc
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc002c29000, {0x6b131a8, 0xc008781e00}, 0xc002c29540, {0x7f525b325e98, 0xc00096ef20})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1253 +0xc13 fp=0xc005168820 sp=0xc0051683f0 pc=0x320e833
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc002c29000, {0x6b131a8?, 0xc008781ce0?}, {{0x6b3c8e0, 0xc007479ef0}, {0x0, 0x0, 0x0}, {0xc006e5120c, 0x153}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:738 +0x2654 fp=0xc005169278 sp=0xc005168820 pc=0x3209d14
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x6b131a8?, 0xc008781ce0?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:130 +0xbc fp=0xc005169328 sp=0xc005169278 pc=0x32070fc
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0x6b131a8?, {0x6b131a8?, 0xc008781ce0?}, {0x6b3c8e0?, 0xc007479ef0?}, 0x0?, 0x9a1f9a0?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2522 +0x330 fp=0xc005169478 sp=0xc005169328 pc=0x3217f50
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc002c29000, {0x6b131a8, 0xc008781ce0}, {{0x6b3c8e0, 0xc007479ef0}, {0x0, 0x0, 0x0}, {0xc006e5120c, 0x153}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129 +0x4fa fp=0xc005169608 sp=0xc005169478 pc=0x3206afa
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0xc005169ac0, 0xc002c29000, 0xc0051697b8, 0xc005169870, 0xc0051698b0, 0xc005169890, 0xc005169880, 0xc0051698a0)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1959 +0x30f fp=0xc0051696d0 sp=0xc005169608 pc=0x31fb86f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc002c29000)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1964 +0x1127 fp=0xc005169cd0 sp=0xc0051696d0 pc=0x31fa187
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc002c29000, {0x6b13100, 0xc002038e40}, 0x410cc9e000000000?, 0xc003985dd8?, 0xc00495ee50?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1882 +0x20b fp=0xc005169d88 sp=0xc005169cd0 pc=0x31f8ccb
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc002265e00?, {0x6b13100?, 0xc002038e40?}, {0x5592ce9?}, 0x3?, 0xc00495a978?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:834 +0xe6 fp=0xc005169e38 sp=0xc005169d88 pc=0x31f4a66
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1()
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:727 +0x40d fp=0xc005169fe0 sp=0xc005169e38 pc=0x415832d
runtime.goexit()
	GOROOT/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc005169fe8 sp=0xc005169fe0 pc=0x4d2a41
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:640 +0x22e

```

--- a/pkg/sql/catalog/randgen/BUILD.bazel
+++ b/pkg/sql/catalog/randgen/BUILD.bazel
@@ -42,6 +42,7 @@ go_library(
         "//pkg/sql/sem/tree",
         "//pkg/sql/sessiondata",
         "//pkg/sql/types",
+        "//pkg/util/mon",
         "//pkg/util/randident",
         "@com_github_cockroachdb_errors//:errors",
     ],

--- a/pkg/sql/catalog/randgen/privs.go
+++ b/pkg/sql/catalog/randgen/privs.go
@@ -46,7 +46,7 @@ func (g *testSchemaGenerator) checkCanCreateOnSchema(
 		}
 
 	default:
-		panic(errors.AssertionFailedf("unknown schema kind %d", sc.SchemaKind()))
+		panic(genError{errors.AssertionFailedf("unknown schema kind %d", sc.SchemaKind())})
 	}
 }
 

--- a/pkg/sql/catalog/randgen/randgen.go
+++ b/pkg/sql/catalog/randgen/randgen.go
@@ -12,8 +12,8 @@ package randgen
 
 import (
 	"context"
-	"math"
 	"math/rand"
+	"unsafe"
 
 	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
@@ -32,6 +32,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/util/mon"
 	"github.com/cockroachdb/errors"
 )
 
@@ -47,6 +48,7 @@ func NewTestSchemaGenerator(
 	currentSearchPath sessiondata.SearchPath,
 	kvTrace bool,
 	idGen eval.DescIDGenerator,
+	mon *mon.BytesMonitor,
 	rand *rand.Rand,
 ) TestSchemaGenerator {
 	dbPrivs := catpb.NewBaseDatabasePrivilegeDescriptor(callingUser)
@@ -62,6 +64,7 @@ func NewTestSchemaGenerator(
 	g.ext.cat = cat
 	g.ext.coll = descCollection
 	g.ext.idGen = idGen
+	g.ext.mon = mon
 
 	g.cfg.TestSchemaGeneratorConfig = cfg
 	g.cfg.user = callingUser
@@ -179,6 +182,11 @@ func (g *testSchemaGenerator) Generate(
 				"only admin users can generate more than %d descriptors at a time", maxNewDescsPerNonAdmin)
 		}
 
+		// Check that we have enough memory upfront.
+		acc := g.ext.mon.MakeBoundAccount()
+		defer acc.Close(ctx)
+		g.checkMemoryUsage(ctx, &acc, expDbs, expScs, expTbs)
+
 		// Where are we creating the objects?
 		g.lookupBaseNamespaces(ctx, tn)
 
@@ -237,6 +245,7 @@ type testSchemaGenerator struct {
 	ext struct {
 		txn *kv.Txn
 		st  *cluster.Settings
+		mon *mon.BytesMonitor
 
 		// Catalog interface.
 		cat Catalog
@@ -302,10 +311,17 @@ func (g *testSchemaGenerator) prepareCounts(
 	namePattern *tree.TableName,
 ) (expectedNewDbs, expectedNewSchemas, expectedNewTables int) {
 	counts := g.cfg.Counts
+	const reasonableMaxCount = 10000000
 	for _, sz := range counts {
-		if sz < 0 || sz > math.MaxInt {
+		// We cap the individual count to a maximum number for two purposes.
+		// One is that abnormally large values are likely indicative of
+		// of a user mistake; in which case we want an informative error message
+		// instead of something related to memory usage.
+		// The second is that we want to avoid an overflow in the computation
+		// of the byte monitor pre-allocation in checkMemoryUsage().
+		if sz < 0 || sz > reasonableMaxCount {
 			panic(genError{errors.WithHintf(pgerror.Newf(pgcode.Syntax, "invalid count"),
-				"Each count must be between 0 and %d.", math.MaxInt)})
+				"Each count must be between 0 and %d.", reasonableMaxCount)})
 		}
 	}
 
@@ -362,7 +378,48 @@ func (g *testSchemaGenerator) prepareCounts(
 	if g.gencfg.createDatabases {
 		numNewTables = numNewTables * g.gencfg.numDatabases
 	}
+	if numNewDatabases+numNewSchemas+numNewTables > reasonableMaxCount {
+		panic(genError{errors.WithHintf(
+			pgerror.Newf(pgcode.Syntax, "too many objects generated"),
+			"Can generate max %d at a time.", reasonableMaxCount)})
+	}
 	return numNewDatabases, numNewSchemas, numNewTables
+}
+
+func (g *testSchemaGenerator) checkMemoryUsage(
+	ctx context.Context, acc *mon.BoundAccount, expDbs, expScs, expTbs int,
+) {
+	// Below we count the descriptors two times, one for the go struct
+	// stored in the catalog and one for the protobuf encoding in the
+	// KV batch.
+	// We also count the length of the name patterns 4 times, because
+	// we estimate that the noise generation can add up to 4 bytes
+	// per byte in the name on average.
+	if err := acc.Grow(ctx,
+		// Descriptors.
+		int64(expDbs)*(int64(2*unsafe.Sizeof(g.models.db))+
+			int64(2*unsafe.Sizeof(g.models.publicSc)))+
+			// Names and IDs.
+			int64(g.gencfg.numDatabases)*(int64(4*len(g.gencfg.dbNamePat))+
+				int64(unsafe.Sizeof(descpb.ID(0))))); err != nil {
+		panic(genError{err})
+	}
+	if err := acc.Grow(ctx,
+		// Descriptors.
+		int64(expScs)*int64(2*unsafe.Sizeof(g.models.sc))+
+			// Names and IDs.
+			int64(g.gencfg.numSchemasPerDatabase)*(int64(4*len(g.gencfg.scNamePat))+
+				int64(unsafe.Sizeof(descpb.ID(0))))); err != nil {
+		panic(genError{err})
+	}
+	if err := acc.Grow(ctx,
+		// Descriptors.
+		int64(expTbs)*int64(2*unsafe.Sizeof(g.models.tb))+
+			// Names and IDs.
+			int64(g.gencfg.numTablesPerSchema)*(int64(4*len(g.gencfg.tbNamePat))+
+				int64(unsafe.Sizeof(descpb.ID(0))))); err != nil {
+		panic(genError{err})
+	}
 }
 
 func (g *testSchemaGenerator) lookupBaseNamespaces(

--- a/pkg/sql/generate_objects.go
+++ b/pkg/sql/generate_objects.go
@@ -58,6 +58,7 @@ func (p *planner) GenerateTestObjects(ctx context.Context, parameters string) (s
 		p.CurrentSearchPath(),
 		p.ExtendedEvalContext().Tracing.KVTracingEnabled(),
 		p.EvalContext().DescIDGenerator,
+		p.Mon(),
 		rng,
 	)
 

--- a/pkg/sql/logictest/testdata/logic_test/gen_test_objects
+++ b/pkg/sql/logictest/testdata/logic_test/gen_test_objects
@@ -551,6 +551,22 @@ SELECT crdb_internal.generate_test_objects('foo', 10)
 statement ok
 RESET CLUSTER SETTING sql.schema.test_object_generator.enabled
 
+subtest max_counts
+
+query error invalid count
+SELECT crdb_internal.generate_test_objects('foo', 100000000)
+
+query error invalid count
+SELECT crdb_internal.generate_test_objects('{"names":"a.b.c","counts":[100000000,0,0]}'::jsonb)
+
+query error invalid count
+SELECT crdb_internal.generate_test_objects('{"names":"a.b.c","counts":[0,100000000,0]}'::jsonb)
+
+query error invalid count
+SELECT crdb_internal.generate_test_objects('{"names":"a.b.c","counts":[0,0,100000000]}'::jsonb)
+
+query error too many objects generated
+SELECT crdb_internal.generate_test_objects('{"names":"a.b.c","counts":[10000000,10000000,10000000]}'::jsonb)
 
 subtest temp_schema
 

